cli/src/cluster_query.rs

```diff
@@ -130,7 +154,77 @@ pub fn parse_show_validators(matches: &ArgMatches<'_>) -> Result<CliCommandInfo,
     })
 }
 
-pub fn process_cluster_version(rpc_client: &RpcClient, _config: &CliConfig) -> ProcessResult {
+/// Creates a new process bar for processing that will take an unknown amount of time
+fn new_spinner_progress_bar() -> ProgressBar {
+    let progress_bar = ProgressBar::new(42);
+    progress_bar
+        .set_style(ProgressStyle::default_spinner().template("{spinner:.green} {wide_msg}"));
+    progress_bar.enable_steady_tick(100);
+    progress_bar
+}
+
+pub fn process_catchup(rpc_client: &RpcClient, node_pubkey: &Pubkey) -> ProcessResult {
+    let cluster_nodes = rpc_client.get_cluster_nodes()?;
+    let contact_info = cluster_nodes
+        .iter()
+        .find(|contact_info| contact_info.pubkey == node_pubkey.to_string());
+    if contact_info.is_none() {
```

Comment:
I think you can rewrite the next 3 statements a little more Rustily, something like:
```
let rpc = contact_info
  .ok_or_else(|| format!("Contact information not found for {}", node_pubkey).into())?
  .rpc
  .ok_or_else(|| format!("RPC service not found for {}", node_pubkey).into())?;
```
I can't decide if that's actually more concise.

Reply:
Sounds like job security, done!